easybuild/easyblocks/m/motioncor2.py

```diff
@@ -0,0 +1,112 @@
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing MotionCor2, implemented as an easyblock
+
+@author: Ake Sandgren, (HPC2N, Umea University)
+"""
+
+import os
+import stat
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, copy_file, mkdir, write_file
+
+
+class EB_MotionCor2(EasyBlock):
+    """
+    Support for installing MotionCor2
+     - creates wrapper that loads the correct version of CUDA before
+     - running the actual binary
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(EB_MotionCor2, self).__init__(*args, **kwargs)
+
+        self.cuda_mod_name, self.cuda_name = None, None
+        for dep in self.cfg.dependencies():
+            if dep['name'] == 'CUDA':
+                self.cuda_mod_name = dep['short_mod_name']
+                self.cuda_name = self.cuda_mod_name.rsplit('/', 1)[0]
+                cuda_ver = dep['version']
+                break
+
+        if self.cuda_mod_name is None:
+            raise EasyBuildError("CUDA must be a direct (build)dependency of MotionCor2")
+        cuda_short_ver = "".join(cuda_ver.split('.')[:2])
+        self.motioncor2_bin = 'MotionCor2_%s-Cuda%s' % (self.version, cuda_short_ver)
+
+    def configure_step(self):
+        """No configuration, this is binary software"""
+        pass
+
+    def build_step(self):
+        """No compilation, this is binary software"""
+        pass
+
+    def install_step(self):
+        """
+        Install binary and a wrapper that loads correct CUDA version.
+        """
+
+        src_mc2_bin = os.path.join(self.builddir, self.motioncor2_bin)
+        if not os.path.exists(src_mc2_bin):
+            raise EasyBuildError("Specified CUDA version has no corresponding MotionCor2 binary")
+
+        bindir = os.path.join(self.installdir, 'bin')
+        mkdir(bindir)
+
+        dst_mc2_bin = os.path.join(bindir, self.motioncor2_bin)
+        copy_file(src_mc2_bin, dst_mc2_bin)
+
+        exe_perms = stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH
```

Comment:
Are we sure that read permissions are enable for all?

Reply:
I assume you intended to make the comment in two places...

Anyway, for the wrapper it should be unless user has set an odd umask that write_file picks up on.
I'll add reads, just forgot it when i changed it...

Reply:
Doesn't hurt to make sure here :)